### ISSUE TITLE
[RESOLVED] Tor problems

### ISSUE THREAD

**Issue by MetallDoktor** [cuckoosandbox/cuckoo#1995]
Hello again!

There is actually one more problem I can't resolve. I set tor routing strictly by the manual (installed tor itself, added necessary changes to [tor] in routing.conf and torrc, restarted tor). But each time I submit a task via tor routing I get this: **It appears that this Virtual Machine hasn't been configured properly as the Cuckoo Host wasn't able to the connect to the Guest or the other way around (i.e., Guest wasn't able to contact the Cuckoo Host).**

Had this error earlier when it was actually some network misconfigurations, but it should not be now (at least, VPN/Inetsim routing works just fine).

- VM Windows 7, windows firewall disabled. VM gets ping responses from cuckoo host during the task processing, agent is on (again, VPN/Inetsim routing on the same machine works as it should be)
- Cuckoo rooter shows messages like this: tor_enable 192.168.56.5 0.0.0.0 5353 9040 (and tor_disable respectively) and adds rules to the cuckoo host firewall during the task processing (192.168.56.5 - VM IP)
- On cuckoo host firewall input policy set as ACCEPT
- In torrc (all other lines commented, tried two ways showed below (with one/two-line), same result):

    #TransPort 192.168.56.1:9040 
    #DNSPort 192.168.56.1:5353 
    TransListenAddress 192.168.56.1                
    TransPort 9040        
    DNSListenAddress 192.168.56.1        
    DNSPort 5353

-  Tor service is running (netstat data below):

    tcp        0      0 127.0.0.1:9040          0.0.0.0:*               LISTEN      2705/tor 
    tcp        0      0 127.0.0.1:9050          0.0.0.0:*               LISTEN      2705/tor 
    udp        0      0 127.0.0.1:5353          0.0.0.0:*                           2705/tor

Also installed tor-arm to check the actual work of tor and get this message: **The torrc differs from what tor's using. You can issue a sighup to reload the torrc values by pressing x. - configuration value is missing from the torrc: RunAsDaemon**. No matter if I'm doing it or not, same result.

And, just in case, data from ps aux | grep tor: /usr/bin/tor --defaults-torrc /usr/share/tor/tor-service-defaults-torrc -f /etc/tor/torrc --RunAsDaemon 0

Thanks in advance!

**Comment by doomedraven** [comment]
as your line says, your /etc/torrc is ignored

do identical both confs `/etc/torrc` and `/usr/share/tor/tor-service-defaults-torrc`
check which tor is used, i know there is like default tor in ubuntu and you install another one from repo

**Comment by MetallDoktor** [comment]
Already, same result, content of this file below:

DataDirectory /var/lib/tor
PidFile /var/run/tor/tor.pid
RunAsDaemon 1
User debian-tor

ControlSocket /var/run/tor/control GroupWritable RelaxDirModeCheck
ControlSocketsGroupWritable 1
SocksPort unix:/var/run/tor/socks WorldWritable
SocksPort 9050

CookieAuthentication 1
CookieAuthFileGroupReadable 1
CookieAuthFile /var/run/tor/control.authcookie

Log notice syslog

TransPort 192.168.56.1:9040
DNSPort 192.168.56.1:5353

**Comment by doomedraven** [comment]
that is very weird, could you provide more info about, os, cuckoo version, tor version, etc

**Comment by MetallDoktor** [comment]
Tor 0.3.1.7 (from syslog)
Cuckoo 2.0.4.4 (from pip show cuckoo)
Ubuntu 17.04 (from lsb_release -a)

Something else? Can't figure out if there is something else that may help. Cuckoo log is not very different from normal except maybe this line: **INFO: cuckoo_test: end of analysis reached!**

**Comment by doomedraven** [comment]
that info from cuckoo is not important, no idea im using `16.04 as it's LTS`
and my tor is `Tor 0.2.9.11 (git-aa8950022562be76) running on Linux with Libevent 2.0.21-stable, OpenSSL 1.0.2g and Zlib 1.2.8.`

cuckoo is the same, so idk what is wrong on your side

**Comment by MetallDoktor** [comment]
Finally, I've worked this out. Maybe someone else has the same problem, so I am writing it here.

When I've just configured cuckoo for the first time, in cuckoo.conf I specified [resultserver] as 0.0.0.0 (for that moment it was not very clear, if it's gonna be the only instance). With task via tor uploading, cuckoo rooter adds respective firewall rules to nat table according to tor_toggle in rooter.py.

In practice, it looks something like this...:

-A PREROUTING -s 192.168.56.5/32 -p tcp -m tcp --dport 53 -j DNAT --to-destination 0.0.0.0:5353
-A PREROUTING -s 192.168.56.5/32 -p udp -m udp --dport 53 -j DNAT --to-destination 0.0.0.0:5353
-A PREROUTING -s 192.168.56.5/32 ! -d 0.0.0.0/32 -p tcp -j DNAT --to-destination 0.0.0.0:9040
-A PREROUTING -s 192.168.56.5/32 ! -d 0.0.0.0/32 -p udp -j DNAT --to-destination 0.0.0.0:9040

... and has no sense.

So I specified [resultserver] strictly as cuckoo host ip and removed this option in virtualbox.conf VM descriptions. 

And all goes juuuust fine. Since that, I am closing this issue.
